File: PyLibs/src/Libs/FrameWork/ProcessSys.py
# -*- coding: utf-8 -*-

'''
@brief 系统循环
'''

from Libs.Core.GObject import GObject;
from Libs.FrameWork.Ctx import Ctx;

class ProcessSys(GObject):
    
    def __init__(self):
        super(ProcessSys, self).__init__();
        
        self.mTypeId = "ProcessSys";


    def ProcessNextFrame(self):
        Ctx.instance().mSystemTimeData.nextFrame();
        self.Advance(Ctx.instance().mSystemTimeData.getDeltaSec());


    def Advance(self, delta):
        Ctx.instance().mSystemFrameData.nextFrame(delta);
        Ctx.instance().mTickMgr.Advance(delta);                # 心跳
        Ctx.instance().mTimerMgr.Advance(delta);               # 定时器
        Ctx.instance().mFrameTimerMgr.Advance(delta);          # 帧定时器

File: JSLib/src/FrameWork/ProcessSys.js
/**
 * @brief 系统循环
 */
namespace SDK.Lib
{
    public class ProcessSys
    {
        public ProcessSys()
        {

        }

        public void ProcessNextFrame()
        {
            Ctx.m_instance.m_systemTimeData.nextFrame();
            Advance(Ctx.m_instance.m_systemTimeData.deltaSec);
        }

        public void Advance(float delta)
        {
            Ctx.m_instance.m_systemFrameData.nextFrame(delta);
            //Ctx.m_instance.m_luaSystem.Advance(delta);            // lua 脚本 Advance
            Ctx.m_instance.m_tickMgr.Advance(delta);            // 心跳
            //Ctx.m_instance.m_timerMgr.Advance(delta);           // 定时器
            //Ctx.m_instance.m_frameTimerMgr.Advance(delta);           // 帧定时器
        }
    }
}